core/store/src/flat/storage.rs

```diff
@@ -283,6 +291,15 @@ impl FlatStorage {
             block_hash,
             CachedFlatStateDelta { metadata: delta.metadata, changes: Arc::new(cached_changes) },
         );
+        let cached_changes_num_items = guard.metrics.cached_changes_num_items.get() as usize;
```

Comment:
please don't use prometheus metrics data source, that is super hacky :) use `cached_changes.len()` and `cached_changes.total_size()` instead

Reply:
Btw this line was even wrong. I wanted to count number of deltas, not number of items in them - fixed.